app/rest/secret.js

```diff
@@ -0,0 +1,10 @@
+angular.module('portainer.rest')
+.factory('Secret', ['$resource', 'Settings', function SecretFactory($resource, Settings) {
+  'use strict';
+  return $resource(Settings.url + '/secrets/:id/:action', {}, {
+    get: { method: 'GET', params: {id: '@id'} },
+    list: { method: 'GET', isArray: true },
```

Comment:
Please replace `list` by `query` to stay coherent with the others REST services.